resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/VolcanoFeatureStep.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.k8s.features
+
+import java.util
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+import io.fabric8.kubernetes.api.model.{HasMetadata, ObjectMeta, PodBuilder, Quantity, QuantityBuilder}
+import sh.volcano.scheduling.{PodGroup, PodGroupSpec, PodGroupStatus, v1beta1}
+import org.apache.spark.deploy.k8s.{KubernetesConf, KubernetesDriverSpecificConf, KubernetesRoleSpecificConf, KubernetesUtils, SparkPod}
+import org.apache.spark.deploy.k8s.Config._
+
+private[spark] class VolcanoFeatureStep(
+                                         kubernetesConf: KubernetesConf[_ <: KubernetesRoleSpecificConf])
+  extends KubernetesFeatureConfigStep {
+
+  private val conf = kubernetesConf.sparkConf
+  private val podGroupName = s"${kubernetesConf.appResourceNamePrefix}-podgroup"
+  private val deployModeKey = "spark.submit.deployMode"
+  private val clusterMode = "cluster"
+
+  override def configurePod(pod: SparkPod): SparkPod = {
```

Comment:
This looks reasonable to me, just more as a note to myself for something I want to explore here is using a replicaset per resource profile.